lib/imagefs.coffee

```diff
@@ -72,17 +88,32 @@ exports.read = (definition) ->
 # , fs.createReadStream('/baz/qux')
 ###
 exports.write = (definition, stream) ->
-	driver.interact(definition.image, definition.partition).then (fat) ->
-		return stream.pipe(fat.createWriteStream(definition.path))
-			.on('close', fat.closeDriver)
+	if _.isString(definition.image)
+		fs.openAsync(definition.image, 'r+')
+		.then (fd) ->
+			close = -> fs.closeAsync(fd)
+			disk = new filedisk.FileDisk(fd, false, false)
+			write(disk, definition.partition, definition.path, stream)
+			.tap (writeStream) ->
+				writeStream.on('close', close)
+				writeStream.on('error', close)
+	else if definition.image instanceof filedisk.Disk
+		write(definition.image, definition.partition, definition.path, stream)
+	else
+		throw new Error('image must be a String or a filedisk.Disk instance')
```

Comment:
Seems like this error message appears in multiple places - perhaps a shared error-handling function / wrapper could be extracted?